src/tests/functional/shared_test_classes/include/shared_test_classes/base/benchmark.hpp

```diff
@@ -0,0 +1,185 @@
+                assert(found_profile != perf_results.end());
+                time += found_profile->second.realTime_uSec;
+            }
+        }
+
+        long long total_us = 0;
+        for (auto& res : results_us) {
+            const std::string node_type_name = res.first;
+            long long& time = res.second;
+            time /= num_attempts_;
+            total_us += time;
+            std::cout << std::fixed << std::setfill('0') << node_type_name << ": " << time << " us\n";
+        }
+        std::cout << std::fixed << std::setfill('0') << "Total time: " << total_us << " us\n";
+    }
+
+private:
+    std::vector<std::string> bench_node_type_names_;
+    std::chrono::milliseconds warmup_time_;
+    int num_attempts_;
+};
+
+}  // namespace LayerTestsDefinitions
+
+namespace ov {
+namespace test {
+
+template <typename BaseLayerTest>
+class BenchmarkLayerTest : public BaseLayerTest {
+    static_assert(std::is_base_of<SubgraphBaseTest, BaseLayerTest>::value,
+                  "BaseLayerTest should inherit from LayerTestsUtils::LayerTestsCommon");
+
+ public:
+    static constexpr int kDefaultNumberOfAttempts = 100;
+
+    void run(const std::initializer_list<std::string>& node_type_names,
+             const std::chrono::milliseconds warmupTime = std::chrono::milliseconds(2000),
+             const int numAttempts = kDefaultNumberOfAttempts) {
+        bench_node_type_names_ = node_type_names;
+        warmup_time_ = warmupTime;
+        num_attempts_ = numAttempts;
+        SubgraphBaseTest::configuration = {{"PERF_COUNT", "YES"}};
+        SubgraphBaseTest::run();
+    }
+
+    void run(const std::string& node_type_name,
+             const std::chrono::milliseconds warmupTime = std::chrono::milliseconds(2000),
+             const int numAttempts = kDefaultNumberOfAttempts) {
+        if (!node_type_name.empty()) {
+            run({node_type_name}, warmupTime, numAttempts);
+        } else {
+            run({}, warmupTime, numAttempts);
+        }
+    }
+
+    // NOTE: Validation is ignored because we are interested in benchmarks results.
+    //       In future the validate method could check if new benchmark results are not worse than previous one
+    //       (regression test), and in case of any performance issue report it in PR.
+    void validate() override {
+    }
+
+ protected:
+    void infer() override {
+        std::map<std::string, long long> results_us{};
+        for (const auto& node_type_name : bench_node_type_names_) {
+            results_us[node_type_name] = {};
+        }
+
+        // Warmup
+        auto warm_current = std::chrono::steady_clock::now();
+        const auto warm_end = warm_current + warmup_time_;
+        while (warm_current < warm_end) {
+            SubgraphBaseTest::infer();
+            warm_current = std::chrono::steady_clock::now();
+        }
+
+        // Benchmark
+        for (int i = 0; i < num_attempts_; ++i) {
+            SubgraphBaseTest::infer();
```

Comment:
Could you please use `SubgraphBaseTest::infer()` once and `inferRequest.infer()` for all subsequent inferences for consistency with `LayerTestsDefinitions::BenchmarkLayerTest`?

Reply:
Done

Reply:
LGTM